src/os/bluestore/BlueStore.h

```diff
@@ -1005,11 +1005,13 @@ class BlueStore : public ObjectStore,
       needs_reshard_begin = needs_reshard_end = 0;
     }
     void request_reshard(uint32_t begin, uint32_t end) {
-      if (begin < needs_reshard_begin) {
-	needs_reshard_begin = begin;
+      if (needs_reshard_end == 0) {
```

Comment:
This should be need_reshard_begin not needs_reshard_end?
Besides zero seems to be a valid value, one should rather use -1 or something to determine if needs_reshard_begin hasn't been set.

Reply:
If `request_reshard(0,0)` is requested, it means we asked to reshard nothing.
In any other case `needs_reshard_end` is != 0. But `needs_reshard_begin` can still be 0 while we have something to reshard.